utils/python/CIME/hist_utils.py

```diff
@@ -0,0 +1,172 @@
+"""
+Functions for actions pertaining to history files.
+"""
+
+from CIME.XML.standard_module_setup import *
+from CIME.XML.component import Component
+
+import logging, glob, os, shutil, re
+
+def _iter_model_file_substrs():
+    drv_comp = Component()
+    models = drv_comp.get_valid_model_components()
+
+    for model in models:
+        model = "cpl" if model == "DRV" else model.lower()
+        yield model
+
+def _get_all_hist_files(testcase, model, suffix="", from_dir=os.getcwd()):
+    suffix = (".%s" % suffix) if suffix else ""
+    test_hists = glob.glob("%s/%s.%s*.h?.nc%s" % (from_dir, testcase, model, suffix))
```

Comment:
I believe that the main point of explicitly dealing with extensions in the previous version was so that we could just find the _last_ history file for each extension. If I understand the new code, it will compare _all_ history files. So, for example, if we run a 5-day test with daily output, then the new version will do 5x as many history file comparisons.

I'm not sure which is better. Certainly the new version is more robust, in that it will catch problems with diagnostic fields that could have slipped through the cracks in the old version. However, it also means greater cost of the cprnc step - I'm going to guess something like 2x - 10x typically, but as high as 100x in a few cases for the CLM test suite, for some long, coarse-resolution tests.

If we do stick with this new version, I think that will argue for:

1. Keeping cprnc in Fortran as opposed to python, because its performance will now be noticeable in more applications; and/or:

2. Decreasing the amount of history output produced by some of the long tests in the test suite.

@mvertens : Do you have any thoughts on this?